src/main/resources/controllers-context.xml

```diff
@@ -28,7 +28,7 @@
         <constructor-arg ref="jdbcDogDao"/>
     </bean>
 
-    <bean id="transactionalDogService" class="com.example.dog.service.TransactionalDogService">
+    <bean id="transactionalDogService" class="com.example.dog.service.TransactionalDogService" primary="true">
```

Comment:
You can nest the original bean here and it won't be exposed and you won't have to set `primary`